include/alpaka/test/queue/QueueCpuOmp2Collective.hpp

```diff
@@ -38,30 +39,35 @@ namespace alpaka
 #        pragma clang diagnostic ignored "-Wweak-vtables"
 #    endif
         //! The CPU collective device queue implementation.
-        class QueueCpuOmp2CollectiveImpl final : public cpu::ICpuQueue
+        struct QueueCpuOmp2CollectiveImpl final : cpu::ICpuQueue
 #    if BOOST_COMP_CLANG
 #        pragma clang diagnostic pop
 #    endif
         {
-        public:
-            QueueCpuOmp2CollectiveImpl(DevCpu const& dev) noexcept : m_dev(dev), m_uCurrentlyExecutingTask(0u)
+            explicit QueueCpuOmp2CollectiveImpl(DevCpu const& dev) noexcept : m_dev(dev), blockingQueue(dev)
             {
             }
-            QueueCpuOmp2CollectiveImpl(QueueCpuOmp2CollectiveImpl const&) = delete;
-            auto operator=(QueueCpuOmp2CollectiveImpl const&) -> QueueCpuOmp2CollectiveImpl& = delete;
+
             void enqueue(EventCpu& ev) final
             {
                 alpaka::enqueue(*this, ev);
             }
+
             void wait(EventCpu const& ev) final
             {
                 alpaka::wait(*this, ev);
             }
 
-        public:
+            void busyWaitUntilBlockingQueueEmpty()
+            {
+                while(!empty(blockingQueue))
+                    std::this_thread::yield();
```

Comment:
Is this safe in terms of OpenMP legalese?

Reply:
I honestly have no idea :D I know back when C++11 was new, people said to not mix `std::thread`s and OpenMP. I don't know whether that is still true. Yielding should be less probablematic I guess. I am even assuming it should not pose a problem at all, since a yield is usually just a hint to processor or OS to reschedule the thread, but otherwise has no side effect.

Reply:
Yes please do not assume that OpenMP is using standard threads. So I suggest not using `std::this_thread::yield()`

Reply:
I replaced it by a busy loop now, as in the original implementation.